### PR TITLE
Feature/charselect improvements

### PR DESCRIPTION
**Pull request by FPaizSF**
Char select now uses randomized names and allows for players to unlock a hero selection or quit to main menu. It also does better at handling two players selecting the same hero and fixes minor visual issues. 
Addresses the following issues / improvements in char select:
GOMPS-390 Multi player ready clicks prevent game start
GOMPS-350  Random name should be used in char select
GOMPS-292  When no one has selected anything in lobby, should still see there are players connected
GOMPS-293  be able to cancel ready button
GOMPS-375  There is no way to go back to the start from the game lobby
GOMPS-403  Char select UI hides characters on right when ready

### CODE REVIEW

**Comment by unity-cla-assistant** [comment]
[CLA assistant check](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.multiplayer.samples.coop?pullRequest=303) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.multiplayer.samples.coop?pullRequest=303) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.cds.internal.unity3d.com/check/Unity-Technologies/com.unity.multiplayer.samples.coop?pullRequest=303) it.</sub>

File: Assets/BossRoom/Scripts/Client/Game/State/ClientCharSelectState.cs [hunk 1]
```diff
@@ -234,18 +234,18 @@ private void UpdateCharacterSelection(CharSelectData.SeatState state, int seatId
             }
             else
             {
-                m_InSceneCharacter.gameObject.SetActive(true);
-                m_InSceneCharacter.SwapToModel(CharSelectData.LobbySeatConfigurations[seatIdx].CharacterArtIdx);
-                m_ClassInfoBox.ConfigureForClass(CharSelectData.LobbySeatConfigurations[seatIdx].Class);
+                if ( seatIdx != -1 )
+                {
+                    m_InSceneCharacter.gameObject.SetActive(true);
+                    m_InSceneCharacter.SwapToModel(CharSelectData.LobbySeatConfigurations[seatIdx].CharacterArtIdx);
```
Comment (Cosmin-B) [inline Assets/BossRoom/Scripts/Client/Game/State/ClientCharSelectState.cs:240]: I can't figure out or not if these lines need to be indented?

**Review by Cosmin-B — COMMENTED** [review]
In general i don't see any breaking changes here.

File: Assets/BossRoom/Scripts/Client/UI/UICharSelectClassInfoBox.cs [hunk 2]
```diff
@@ -31,7 +31,9 @@ public class UICharSelectClassInfoBox : MonoBehaviour
         [SerializeField]
         private Image m_Skill3;
         [SerializeField]
-        private Button m_ReadyButton;
+        private Image m_ReadyButtonImg;
```
Comment (fernando-cortez) [inline Assets/BossRoom/Scripts/Client/UI/UICharSelectClassInfoBox.cs:34]: Just to remove abbreviation.

**Review by fernando-cortez — COMMENTED** [review]
Super small changes! Looks great. Lots of improvements here.

File: Assets/BossRoom/Scripts/Server/Game/State/ServerCharSelectState.cs [hunk 3]
```diff
@@ -8,15 +8,21 @@ namespace BossRoom.Server
...
     {
         public override GameState ActiveState { get { return GameState.CharSelect; } }
         public CharSelectData CharSelectData { get; private set; }
 
+        private GameNetPortal m_NetPortal;
+        private ServerGameNetPortal m_ServerNetPortal;
+
         private void Awake()
         {
             CharSelectData = GetComponent<CharSelectData>();
```
[the remark this answers is not in the document]
Comment (fernando-cortez) [reply Assets/BossRoom/Scripts/Server/Game/State/ServerCharSelectState.cs:23]: Closing this as it's not part of this scene.

[the remark this answers is not in the document]
Comment (FPaizSF) [reply Assets/BossRoom/Scripts/Server/Game/State/ServerCharSelectState.cs:21]: As discussed, we will leave this as something set in Awake but I did remove the unneeded m_NetPortal and the extra call to GetComponent

**Review by fernando-cortez — DISMISSED** [review]

**Review by Cosmin-B — DISMISSED** [review]
Looks good

**Comment by FPaizSF** [comment]
> There seems to be a conflict with the Third Party Notices (somehow), so we need that fixed before merging this

OK - this should be resolved now. Pls approve again when you can.

**Review by Cosmin-B — DISMISSED** [review]

**Comment by Cosmin-B** [comment]
@FPaizSF this has some conflicts again since we merged some other PR's

**Review by fernando-cortez — APPROVED** [review]

**Review by Cosmin-B — APPROVED** [review]

**Pull request merged**